portfolio/src/test/java/com/google/sps/CampaignServletTest.java

```diff
@@ -0,0 +1,162 @@
+ 
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import static org.mockito.Mockito.*;
+import java.io.*;
+import java.util.ArrayList;
+import com.google.sps.servlets.GetCampaignsServlet;
+import com.google.sps.data.DatastoreRetrieval;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Before;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v3.services.GoogleAdsServiceClient;
+import com.google.auth.Credentials;
+import com.google.ads.googleads.v3.services.SearchGoogleAdsStreamRequest;
+import com.google.ads.googleads.v3.services.SearchGoogleAdsStreamResponse;
+
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.support.membermodification.MemberMatcher.method;
+
+
+import org.mockito.Spy;
+import org.mockito.InjectMocks;
+
+@PowerMockIgnore("jdk.internal.reflect.*")
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({DatastoreRetrieval.class, GetCampaignsServlet.class, SearchGoogleAdsStreamRequest.class,
+	SearchGoogleAdsStreamResponse.class, GoogleAdsServiceClient.class})
+public final class CampaignServletTest {
+
+  private GetCampaignsServlet servlet = PowerMockito.spy(new GetCampaignsServlet());
```

Comment:
Now that you have a mock for GetCampaignsServlet (TestGetCampaignsServlet), do you want to use the mock class instead?